### PR TITLE
Move to Github Actions

### PR DESCRIPTION
**Pull request by dralley**
closes: #7881
https://pulp.plan.io/issues/7881

### CODE REVIEW

File: dev_requirements.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+coveralls   # For `make test-coverage`
+pre-commit  # For `make auto-format-code`
+sphinx      # For `make docs-html` and `make docs-clean`
+wheel       # For `make package`
+twine       # For `make publish`
+black       # For pre-commit hooks
```
Comment (daviddavis) [inline dev_requirements.txt:6]: Why were these dependencies broken out into a separate file?

Comment (dralley) [reply dev_requirements.txt:6]: Consistency and sensibleness.  It doesn't really make a lot of sense to put development dependencies in "extras_require", that's really meant more for additional features and the like.

**Review by daviddavis — APPROVED** [review]

**Comment by dralley** [comment]
Going to merge and then test

**Comment by dralley** [comment]
@daviddavis Actually I can't merge. Can you?

**Pull request merged**